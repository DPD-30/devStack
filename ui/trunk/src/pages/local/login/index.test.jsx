import React from 'react'
import { renderToString } from 'react-dom/server'
import { HashRouter as Router } from 'react-router-dom'
import { mount } from 'enzyme'


import TestComponent from './index.jsx'

jest.unmock('./index.jsx')

 
describe('', () => {
    it('Login component matching snapshot', () => {
        const comp = renderToString(
            <Router>
                <TestComponent />
            </Router>
        )

        expect(comp).toMatchSnapshot()
    })
})

it ('resets', () => {
    const test = mount(
        <Router>
            <TestComponent />
        </Router>
    )
    const comp = test.find(TestComponent)
    const btnReset = comp.find('button.reset')

    btnReset.simulate('click')

    expect(comp.instance().state.loginEmail).toBe('')
    expect(comp.instance().state.loginPassword).toBe('')
    expect(comp.instance().state.formSubmitted).toBeFalsy()
})

it ('changes email field', () => {
    const test = mount(
        <Router>
            <TestComponent />
        </Router>
    )
    const comp = test.find(TestComponent)
    const inputField = comp.find('input[name="email"]')

    inputField.simulate('change', { target: { name: 'loginEmail', value: 'mock@test.com' }})

    expect(comp.instance().state.loginEmail).toBe('mock@test.com')
})

it ('calls log in function', () => {
    const test = mount(
        <Router>
            <TestComponent />
        </Router>
    )
    const comp = test.find(TestComponent)
    const btnLogin = comp.find('button.login')

    comp.setState({ email: 'mock@test.com', password: 'Test123!' })

    btnLogin.simulate('click')

    // expect(comp.instance().state.loginEmail).toBe('')
})

it ('calls log in function with blanks', () => {
    const test = mount(
        <Router>
            <TestComponent />
        </Router>
    )
    const comp = test.find(TestComponent)
    const btnLogin = comp.find('button.login')

    comp.setState({ email: '', password: '' })

    btnLogin.simulate('click')

    // expect(comp.instance().state.loginEmail).toBe('')
})

it ('show login failuire if failed', () => {
    const test = mount(
        <Router>
            <TestComponent />
        </Router>
    )
    const comp = test.find(TestComponent)

    comp.setState({ localUser_status: 'FAILURE' })

    // expect(comp.instance().state.loginEmail).toBe('')
})
