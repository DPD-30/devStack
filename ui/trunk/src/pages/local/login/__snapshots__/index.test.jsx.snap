// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component matching snapshot 1`] = `
<div class="card">
  <div class="card-body">
    <div>
      <h2>
        Login
      </h2>
      <br>
      <br>
      <div class="form-group">
        <label for="email"
               class
        >
          Email Address
        </label>
        <input type="email"
               name="email"
               id="email"
               placeholder
               value
               class="form-control"
               aria-invalid="false"
        >
        <div class="invalid-feedback">
          Email address is required
        </div>
      </div>
      <br>
      <div class="form-group">
        <label for="password"
               class
        >
          Password
        </label>
        <input type="password"
               name="password"
               id="password"
               placeholder
               value
               class="form-control"
               aria-invalid="false"
        >
        <div class="invalid-feedback">
          Password is required
        </div>
      </div>
      <div class="justify-content-center">
        <button type="button"
                class="reset btn btn-secondary btn-lg"
        >
          Reset
        </button>
        <button type="button"
                style="margin-left:20px"
                class="login btn btn-primary btn-lg"
        >
          Login
        </button>
      </div>
      <br>
      <hr>
      <p>
        <a href="#/register">
          <button class="btn btn-link">
            Register
          </button>
        </a>
      </p>
    </div>
  </div>
</div>
`;
